generics/map.go

```diff
@@ -0,0 +1,20 @@
+//go:build go1.20
+
+package generics
+
+// MapSlice transforms a slice of a given type to new slice of a different type
+func MapSlice[T, K any](ss []T, fn func(v T) (K, error)) ([]K, error) {
```

Comment:
do we want the error in there should we let the user deal with  it on its own? because if you get out an error, why not getting in a context?

Reply:
Having an error can prove most useful when there is parsing involved just like a `strconv.ParseInt`. We can indeed add a context here.